src/App.tsx

```diff
@@ -1,29 +1,47 @@
 import * as React from "react";
-import { BrowserRouter as Router, Route, Link } from "react-router-dom";
-import styled from "styled-components";
-import {GlobalStyle} from "./global";
-import "./App.css";
-import NavBar from "./components/NavBar/NavBar"
-import Home from "./pages/Home/Home"
-import Restaurants from "./pages/Restaurants/restaurantsList"
-import dishPreview from "./pages/Restaurants/dishPreview"
-import Footer from "./components/Footer/footer"
+import { useState, useEffect } from "react";
 
+import { BrowserRouter as Router, Route } from "react-router-dom";
+import firebase from "./firebase";
+// import styled from "styled-components";
+import { GlobalStyle } from "./global";
+import "./App.css";
+import NavBar from "./components/NavBar/NavBar";
+import Home from "./pages/Home/Home";
+import Restaurants from "./pages/Restaurants/restaurantsList";
+import Dishes from "./pages/Dishes/DishList";
+// import dishPreview from "./pages/Restaurants/dishPreview";
+import Footer from "./components/Footer/footer";
 
 function App() {
+  const [restaurants, setRestaurants] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const db = firebase.firestore();
+      let restaurantsRef = db.collection('restaurants');
+      let allRestaurants = await restaurantsRef.get()
+      let dataArray:any = []
+      for (const doc of allRestaurants.docs){
+        dataArray.push({...doc.data(), id: doc.id})
+      }
+      setRestaurants(dataArray)
+    };
+    
+    fetchData();
+  },[]);
+
   return (
     <Router>
-          <GlobalStyle/>
-    <div className="App">
-      <NavBar />
-   <Route path="/" exact component = {Home} />
-   <Route path="/restaurants" component = {Restaurants} />
-   <Route path="/restaurants/:id" component = {Restaurants} />
-   <Footer /> 
-
-    </div>
+      <GlobalStyle />
+      <div className="App">
+        <NavBar />
+        <Route path="/" exact component={Home} />
+        <Route path="/restaurants" exact component={Restaurants} restaurants={restaurants} />
```

Comment:
As a follow up to my previous comment about fetching data, don't pass data around in props like this, this will work but is generally frowned upon. Think maybe that you might need some data that is in the restaurants inside the home page, dish page or chef page, you will have to pass props to each and every one of these components..
A global state management library like Redux could help out a ton here.

Reply:
I'm pretty sure this doesn't affect the app at all, I just forgot to comment out this part when I was testing around with firebase.

Reply:
But using a tool like redux is probably a good solution anyways.